package/MDAnalysis/lib/log.py

```diff
@@ -394,3 +396,34 @@ def echo(self, step, **kwargs):
             return
         echo(self.format_handler(format, vars(self)),
              replace=self.dynamic, newline=newline)
+
+
+# detect if running from jupyter, ipython or python
+try:
+    shell = get_ipython().__class__.__name__
+except: # non-interactive shell
```

Comment:
Can you be more precise about the exception that is raised, e.g., `NameError`?

Some comment on how you do the detection would make it clearer.

Reply:
I noticed that tqdm actually has a class for automatic detection of this so I'll be changing the code to use their solution instead